### ISSUE TITLE
Créer un environnement de staging

### ISSUE THREAD

**Issue by florianpanchout** [betagouv/recosante-api#264]

**Comment by florianpanchout** [comment]
D'ailleurs si cet environnement peut avoir ses propres templates de mail sur Sendinblue, ce serait chouette.

C'est stressant d'être obligé de faire toutes mes modifications en live aha

**Comment by l-vincent-l** [comment]
L’environnement est accessible ici http://staging.api.recosante.beta.gouv.fr

**Comment by florianpanchout** [comment]
Parfait merci !

**Comment by florianpanchout** [comment]
J'ai une erreur en essayant d'accéder à https://staging.api.recosante.beta.gouv.fr/v1/?insee=75056

**Comment by l-vincent-l** [comment]
Non, c’était moi, l’import des données depuis la prod ne fonctionnait pas

**Comment by florianpanchout** [comment]
@l-vincent-l 

Je pense que notre conversation de tout à l’heure découle du fait que l’on n’attends pas la même chose de cet environnement. Par abus de langage, je vois l’env de staging comme un env de test, alors que tu le vois (très logiquement) vraiment comme un env de staging (je prends https://en.wikipedia.org/wiki/Deployment_environment pour référence sur les termes et fonctions).

La notre environnement est un croisement entre les deux. Et pour notre produit, je me demande si c’est la meilleure solution / la plus simple à maintenir. Est ce qu’avoir 3 environnement testing / staging / production ne serait pas le plus simple ?

- **testing** : tous les process de la prod (récupération des données et envoi de mail inclus) sont répliqués. Par contre la base d’utilisateur n’est pas liée à la base de prod. Une personne inscrite sur l’environnement de prod n’est absolument pas liée à un compte sur l’environnement de test
- **staging** : copie conforme de la prod avec les process couteux (ou visibles comme l’envoi de mail) désactivés. Les utilisateurs de la prod sont répliqués sur cet environnement
- **production** : la prod

Ça permettrais de répondre à nos deux besoins :
- Avoir un environnement sur lequel pouvoir tester l’intégralité des process de la prod (envoi quotidien de mail et récupération des indicateurs inclus) 
- Avoir un environnement le plus iso possible avec la prod (même nombre et diversité d’utilisateurs)

Effectivement ça ne permet pas de tester l'envoi de mail grandeur nature, mais je n'ai pas de solution à ça. 

Qu’en penses-tu ?

**Comment by l-vincent-l** [comment]
Yep on peut faire ça.
Mais je peux monter un env de test; mais ça te va de déplacer sa mise en oeuvre après la JNQA ?

**Comment by florianpanchout** [comment]
Oui totalement. Désolé il s'est automatiquement mis dans priorité quand je l'ai réouvert
